ev3dev2/motor.py

```diff
@@ -1055,6 +1095,9 @@ class MediumMotor(Motor):
 
     def __init__(self, address=None, name_pattern=SYSTEM_DEVICE_NAME_CONVENTION, name_exact=False, **kwargs):
 
+        if platform in ('brickpi', 'brickpi3'):
+            raise Exception("%s is unaware of different motor types, use LargeMotor instead" % platform)
```

Comment:
Is that doable? I have never tried it. I added the snippet above after trying to use a medium motor on brickpi3 but hit issues and dlech mentioned that it cannot detect different motor types

Reply:
I don't know for sure. My assumption was that you could manually set the driver and it would work, it was just that by default the ports were all initialized as large motors for simplicity.

Reply:
I think if someone wants to test/document that then we can drop the exception at that point.

Reply:
I would prefer to not introduce the exception given that it isn't an error case in existing releases. If someone depends on this class, I don't see a reason to break them.

Reply:
My memory was rusty but after going back and re-reading https://github.com/ev3dev/ev3dev/issues/1059 I do not think "set the port driver to the medium motor class explicitly" will work (see the first comment in that issue). So it is an error case today if the user attempts to use the MediumMotor class on a brickpi3.  

This change isn't needed as part of MoveDifferential though so I opened #543 to track it....will drop this part of the diff from the patch here.